backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/domain/model/ItemStock.java

```diff
@@ -0,0 +1,299 @@
+
+package org.eclipse.tractusx.puris.backend.stock.domain.model;
+
+import jakarta.persistence.Embeddable;
+import jakarta.persistence.EmbeddedId;
+import jakarta.persistence.Entity;
+import lombok.*;
+import org.eclipse.tractusx.puris.backend.stock.logic.dto.itemstocksamm.DirectionCharacteristic;
+import org.eclipse.tractusx.puris.backend.stock.logic.dto.itemstocksamm.ItemUnitEnumeration;
+
+import java.io.Serializable;
+import java.util.Date;
+import java.util.Objects;
+
+@NoArgsConstructor
+@Entity
+@ToString
+public class ItemStock {
+
+    @Getter
+    @EmbeddedId
+    private Key key = new Key();
+    private Date lastUpdatedOnDateTime;
+
+    private QuantityOnAllocatedStock quantityOnAllocatedStock = new QuantityOnAllocatedStock();
+    @Getter
+    @Setter
+    private boolean isBlocked;
+
+    public String getPartnerBpnl() {
+        return key.partnerBpnl;
+    }
+
+    public String getMaterialNumberCustomer() {
+        return key.materialNumberCustomer;
+    }
+
+    public String getMaterialNumberSupplier() {
+        return key.materialNumberSupplier;
+    }
+
+    public String getMaterialGlobalAssetId() {
+        return key.getMaterialGlobalAssetId();
+    }
+
+    public DirectionCharacteristic getDirection() {
+        return key.direction;
+    }
+
+    public String getSupplierOrderId() {
+        return key.supplierOrderId;
+    }
+
+    public String getCustomerOrderId() {
+        return key.customerOrderId;
+    }
+
+    public String getCustomerOrderPositionId() {
+        return key.customerOrderPositionId;
+    }
+
+    public String getLocationBpna() {
+        return key.locationBpna;
+    }
+
+    public String getLocationBpns() {
+        return key.locationBpns;
+    }
+
+    public Date getLastUpdatedOnDateTime() {
+        return (Date) lastUpdatedOnDateTime.clone();
+    }
+
+    public void setLastUpdatedOnDateTime(Date lastUpdatedOnDateTime) {
+        this.lastUpdatedOnDateTime = lastUpdatedOnDateTime;
+    }
+
+    public ItemUnitEnumeration getMeasurementUnit() {
+        return quantityOnAllocatedStock.getMeasurementUnit();
+    }
+
+    public Double getQuantityAmount() {
+        return quantityOnAllocatedStock.quantity;
+    }
+
+    public void setQuantityAmount(double quantityAmount) {
+        quantityOnAllocatedStock.quantity = quantityAmount;
+    }
+
+    public void setQuantityOnAllocatedStock(QuantityOnAllocatedStock quantityOnAllocatedStock) {
+        this.quantityOnAllocatedStock = quantityOnAllocatedStock;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof ItemStock)) return false;
+        ItemStock itemStock = (ItemStock) o;
+        return key.equals(itemStock.key);
+    }
+
+    @Override
+    public int hashCode() {
+        return key.hashCode();
+    }
+
+    @Embeddable
+    @Getter
+    @Setter
+    @ToString
+    public static class Key implements Serializable {
+
+        private String partnerBpnl;
```

Comment:
When using an embeddedId as key, the key can't have nullable members. Otherwise there will be SQL-constraint violation errors. And this cannot be mitigated via Hibernate-Annotations. The solution would be either 

- represent nullable values within the entity as empty Strings. Since an empty String can't be a valid value for things like for example supplier order id and all other columns, this seems like an elegant solution to me. And it's not difficult to handle during the process of mapping an ItemStockEntity to a SAMM-Dto. 

- remove all field, that might be nullable from the key alltogether. That's also quite easy to implement, although I personally would prefer to preserve the possibility to differentiate between ItemStockEntities that have for instance an entry for customerOrderId and others that don't.